Source/Core/DolphinNoGUI/Platform.h

```diff
@@ -0,0 +1,24 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <string>
+
+class Platform
+{
+public:
+  virtual void Init() {}
+  virtual void SetTitle(const std::string& title) {}
+  virtual void MainLoop()
```

Comment:
This should go in the cpp file. If this ever changes, everything that includes this file will need to be rebuilt as is.

`Core/Core.h`, `<chrono>` and `<thread>` would also need to be included.